backend/dataall/modules/worksheets/api/resolvers.py

```diff
@@ -69,3 +70,49 @@ def run_sql_query(context: Context, source, environmentUri: str = None, workshee
 def delete_worksheet(context, source, worksheetUri: str = None):
     with context.engine.scoped_session() as session:
         return WorksheetService.delete_worksheet(session=session, uri=worksheetUri)
+
+
+@is_feature_enabled('modules.worksheets.features.nlq')
+def text_to_sql(
+    context: Context,
+    source,
+    environmentUri: str = None,
+    worksheetUri: str = None,
+    prompt: str = None,
+    databaseName: str = None,
+    tableNames: list = None,
+):
+    with context.engine.scoped_session() as session:
```

Comment:
Same comment as before, let's avoid using the passed context object and start the session "too soon". Instead use get_context inside the service layer. Same for analyze_text_genai

Reply:
yup I will fix here and for other worksheet resolvers to match